firewood/src/v2/api.rs

```diff
@@ -64,6 +64,11 @@ pub enum Error {
     #[error("Incorrect root hash for commit: {provided:?} != {current:?}")]
     IncorrectRootHash { provided: HashKey, current: HashKey },
 
+    /// Invalid range
+    /// TODO how to log the range?
+    #[error("Invalid range")]
```

Comment:
We can do a lot better reporting the exact errror here:
```suggestion
    #[error("first_key ({first_key}) must be <= last_key ({last_key})")]
```
[Check out my talk about this](https://medium.com/@rkuris/good-apis-cd861b8b70a3) in particular the section "Solid error reporting" close to the end